SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LayeredLBStrategy.cpp

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/LayeredLBStrategy.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include "Utils/LayerInfo.h"
+#include "Utils/SpatialActorUtils.h"
+
+#include "Templates/Tuple.h"
+
+DEFINE_LOG_CATEGORY(LogLayeredLBStrategy);
+
+ULayeredLBStrategy::ULayeredLBStrategy()
+	: Super()
+{
+}
+
+ULayeredLBStrategy::~ULayeredLBStrategy()
+{
+	for (const auto& Elem : LayerNameToLBStrategy)
+	{
+		Elem.Value->RemoveFromRoot();
+	}
+}
+
+void ULayeredLBStrategy::Init()
+{
+	Super::Init();
+
+	VirtualWorkerId CurrentVirtualWorkerId = SpatialConstants::INVALID_VIRTUAL_WORKER_ID + 1;
+
+	if (const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>())
+	{
+		check(Settings->bEnableUnrealLoadBalancer);
+		const ASpatialWorldSettings* WorldSettings = GetWorld() ? Cast<ASpatialWorldSettings>(GetWorld()->GetWorldSettings()) : nullptr;
+
+		if (WorldSettings == nullptr)
+		{
+			UE_LOG(LogLayeredLBStrategy, Error, TEXT("If EnableUnrealLoadBalancer is set, WorldSettings should inherit from SpatialWorldSettings to get the load balancing strategy."));
+			UAbstractLBStrategy* DefaultLBStrategy = NewObject<UGridBasedLBStrategy>(this);
+			AddStrategyForLayer(SpatialConstants::DefaultLayer, DefaultLBStrategy);
+			return;
+		}
```

Comment:
I think we should catch this much earlier in the dev flow, i.e. when you press play, if load balancer is enabled but you don't have spatial world settings, you're preventing from continuing, and vice versa. This is similar to what we have for other settings iirc. We could keep this here as a sanity check but often I've found misconfiguration can result in some many printed errors that the actual useful error log, e.g. the one you have here, becomes harder to find  and hence diagnose. Also, catching this issue earlier is the flow is always going to be nicer imo.

Reply:
I agree. This had just been moved here from SpatialNetDriver to simplify the code there. Let me look into this in the next PR, when I make changes to the settings, because that's a more natural place to check this.